### PR TITLE
Quick pass through page to update docs

### PR DESCRIPTION
**Pull request by white238**
Noticed the config-build page of the docs was pretty out of date.  This may not be everything.

https://axom.readthedocs.io/en/docs-white238-quickstart/docs/sphinx/quickstart_guide/config_build.html

### CODE REVIEW

**Review by kennyweiss — APPROVED** [review]
Thanks @white238 -- much improved, if not yet perfect!

File: src/docs/sphinx/quickstart_guide/config_build.rst [hunk 1]
```diff
@@ -298,6 +295,8 @@ Here are the key build system options in Axom.
 +------------------------------+---------+--------------------------------+
```
[the remark this answers is not in the document]
Comment (white238) [reply src/docs/sphinx/quickstart_guide/config_build.rst:295]: too internal imo

File: src/docs/sphinx/quickstart_guide/config_build.rst [hunk 2]
```diff
@@ -39,21 +41,23 @@ Compilers we support (listed with minimum supported version):
   * Microsoft Visual Studio 2015
```
[the remark this answers is not in the document]
Comment (agcapps) [reply src/docs/sphinx/quickstart_guide/config_build.rst:41]: @kennyweiss , I build with MS Visual Studio 2017.  I know that some time Al and I will upgrade to a new version of VS. This is related to what VisIt does, b/c one of the build configs uses their libs for debugging.

**Review by agcapps — APPROVED** [review]
Thank you!

**Review by rhornung67 — APPROVED** [review]
Do we really want to maintain a list of compilers we support?

**Comment by white238** [comment]
> 
> 
> Do we really want to maintain a list of compilers we support?

I would prefer to remove it since it's more than likely wrong

**Pull request merged**